primitives/runtime/src/generic/unchecked_extrinsic.rs

```diff
@@ -316,6 +317,20 @@ where
 	}
 }
 
+impl<Address, Call, Signature, Extra> From<UncheckedExtrinsic<Address, Call, Signature, Extra>>
+	for OpaqueExtrinsic
+where
+	Address: Encode,
+	Signature: Encode,
+	Call: Encode,
+	Extra: SignedExtension,
+{
+	fn from(extrinsic: UncheckedExtrinsic<Address, Call, Signature, Extra>) -> Self {
+		OpaqueExtrinsic::decode(&mut extrinsic.encode().as_slice())
+			.expect("decoding from something we have encoded cannot fail; qed")
```

Comment:
The proof is not really correct. The reason it works is not because we just `encoded` it, but rather because both `OpaqueExtrinsic` and `UncheckedExtrinsic` have encoding that is compatible with raw `Vec<u8>` encoding.

Reply:
done